nova/virt/vmwareapi/images.py

```diff
@@ -592,3 +600,22 @@ def upload_image_stream_optimized(context, image_id, instance, session,
 
     LOG.debug("Uploaded image %s to the Glance image server", image_id,
               instance=instance)
+
+
+def ensure_valid_template_vm(session, templ_vm_ref, vmdk_size):
+    """Deletes an invalid template VM by checking the vmdk_size."""
+    if vmdk_size == INVALID_VMDK_SIZE:
+        try:
+            LOG.warning("Deleting invalid template VM %s",
+                        vutil.get_moref_value(templ_vm_ref))
+            destroy_task = session._call_method(
+                session.vim,
+                "Destroy_Task",
+                templ_vm_ref)
+            session._wait_for_task(destroy_task)
+        except Exception:
+            LOG.exception("Failed to delete the template VM %s",
+                          vutil.get_moref_value(templ_vm_ref))
+        return False
+    else:
+        return True
```

Comment:
Optional:
I think we could ditch the `else:` as it serves no purpose.

We could also turn the whole thing around and create less indentation.
```python
if vmdk_size != INVALID_VMDK_SIZE:
    return True

try:
…
```